fdbserver/DeltaTree.h

```diff
@@ -25,13 +25,27 @@
 #include "fdbclient/FDBTypes.h"
 #include "fdbserver/Knobs.h"
 #include <string.h>
+#include <emmintrin.h>
 
 typedef uint64_t Word;
 // Get the number of prefix bytes that are the same between a and b, up to their common length of cl
 static inline int commonPrefixLength(uint8_t const* ap, uint8_t const* bp, int cl) {
 	int i = 0;
+
+	for (; i + 16 < cl; i += 16) {
+		__m128i a = _mm_loadu_si128((__m128i*)(ap + i));
+		__m128i b = _mm_loadu_si128((__m128i*)(bp + i));
+		int r = _mm_movemask_epi8(_mm_cmpeq_epi8(a, b));
```

Comment:
I think you're looking for this intrinsic instead:

`_mm_cmpistri`

it returns the index of the mismatch as an int and knows about null terminators, so you don't need the cl parameter any more.  There's a variant that doesn't know about null terminators, which you might need to use instead (not sure how this function is invoked).

Reply:
Ok, it looks like this is the one you want:

```
_mm_cmpestri
```

It takes the lengths of both strings, so you could pass in cl twice, and have the caller deal with it, or just pass both lengths in, as intended.   https://software.intel.com/sites/landingpage/IntrinsicsGuide/#!=undefined&cats=String%252525252520Compare&expand=956,834

Reply:
> knows about null terminators, so you don't need the cl parameter any more.

We can't assume these are c-style null-terminated strings

Reply:
The E in `_mm_cmpEstri` means it takes explicit string lengths.

For completeness: The "i" means it returns an index as an integer.  "str" means it knows about string types, and "cmp" means compare.